### PR TITLE
[Feature] Add gat_encoder

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR adds a GATEncoder and a test case in test_gnn. 

This PR does not handle 0-in-degree case. As DGL warning, this may lead to invalid results. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/model/gat_encoder.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+        return {'_N': h_conv}
+
+class GATEncoder(GraphConvEncoder):
+    r""" GAT Conv Encoder
+
+    The GATEncoder employs several GATConv Layers as its encoding mechanism.
+    The GATEncoder should be designated as the model's encoder within Graphstorm.
+
+    Parameters
```
Comment (classicsong) [inline python/graphstorm/model/gat_encoder.py:113]: Usually, we put Parameters at the end.

File: python/graphstorm/model/gat_encoder.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        inputs : dict["_N", torch.Tensor]
+            Node feature for each node type.
+        Returns
+        -------
+        dict{"_N", torch.Tensor}
+            New node features for each node type.
+        """
+        g = g.local_var()
+
+        inputs = inputs['_N']
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply python/graphstorm/model/gat_encoder.py:98]: Added.

File: python/graphstorm/model/gat_encoder.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        Number of layers of ngnn between gnn layers
+    ffn_actication: torch.nn.functional
+        Activation Method for ngnn
+    """
+    def __init__(self,
+                 in_feat,
+                 out_feat,
+                 num_heads,
+                 activation=F.relu,
+                 dropout=0.0,
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply python/graphstorm/model/gat_encoder.py:68]: DGL's GATConv has dropout for both features and attentions. And the ffn module also need dropout.

So, it is better to have a dropout argument here.

File: python/graphstorm/model/gat_encoder.py (hunk 1 above)
Comment (zhjwy9343) [reply python/graphstorm/model/gat_encoder.py:113]: changed.

**Review by classicsong — APPROVED** [review]
LGTM
